### DISCUSSION TITLE
Return from update is mismatch

### DISCUSSION THREAD

**Discussion by Mofiqul** [prisma/prisma#14083] [Q&A]
```ts
const user = await prisma.user.update({
  where: { id: 1 },
  data: { email: 'alice@prisma.io' },
})
```

Here the type of `user` is `User` but, if I sent the `user` as JSON response, I get 

```json
{
    "count": 1
}

``` 

What is wrong, I did? Please help

**Comment by austincrim** [comment]
Hey @Mofiqul 👋,

Could you provide the full application code where you're seeing this?

Could you provide your Prisma versions (`npx prisma -v`)?

**Reply by Mofiqul** [reply]
### UserDb 
```ts
        async update(id: string, user: Partial<User>): Promise<Result<User, Error>> { 
             try { 
  
                 const updatedUser = await prisma.user.update({ 
                     where: { id: id }, 
                     data: user 
                 }) 
  
                 return ok(updatedUser); 
  
             } catch (error) { 
                 if (error instanceof Prisma.PrismaClientKnownRequestError) { 
  
                     if (error.code === "P2015") 
                         return err(new Error("user could not be found")) 
  
                     if (error.code === "P2002") 
                         return err(new Error("email already used")) 
  
                 } 
                 throw error; 
             } 
         }
```

**Reply by Mofiqul** [reply]
Using here
```ts
function makeUpdateUser(userDb: UserDb): UseCaseUpdateUser { 
     return async (id: string, data: UserInput) => { 
  
         const { error, value } = UserDataValidation.validate(data); 
  
         if (error) return err(error); 
  
         const input: Partial<User> = updateGuard(value) 
  
         const result = await userDb.update(id, input); 
  
         return result; 
     }
```

**Reply by austincrim** [reply]
What are the contents of `userDb.update`?

**Reply by vladi-stevanovic** [reply]
👋 Hello @Mofiqul  I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
